drift/test/manager/manager_referenced_filter_test.dart

```diff
@@ -516,4 +274,478 @@ void main() {
+            .then((value) => value.length),
+        completion(9));
+  });
+
+  test(
+      'manager - filter related with custom type for primary key with references',
+      () async {
+    // Equals
+    expect(
+        db.managers.todosTable
+            .filter((f) => f.category.id(_todoCategoryData[0].id))
+            .withReferences()
+            .get(distinct: true)
+            .then((value) => value.length),
+        completion(4));
+
+    // Not Equals
+    expect(
+        db.managers.todosTable
+            .filter((f) => f.category.id.not(_todoCategoryData[0].id))
+            .withReferences()
+            .get(distinct: true)
+            .then((value) => value.length),
+        completion(4));
+
+    // Multiple filters
+    expect(
+        db.managers.todosTable
+            .filter((f) => f.category.id(
+                  _todoCategoryData[0].id,
+                ))
+            .filter((f) => f.status.equals(TodoStatus.open))
+            .withReferences()
+            .get(distinct: true)
+            .then((value) => value.length),
+        completion(2));
+
+    // Multiple use related filters twice
+    expect(
+        db.managers.todosTable
+            .filter((f) =>
+                f.category.priority(CategoryPriority.low) |
+                f.category.descriptionInUpperCase("SCHOOL"))
+            .withReferences()
+            .get(distinct: true)
+            .then((value) => value.length),
+        completion(8));
+
+    // Use .filter multiple times
+    expect(
+        db.managers.todosTable
+            .filter((f) => f.category.priority.equals(CategoryPriority.high))
+            .filter((f) => f.category.descriptionInUpperCase("SCHOOL"))
+            .withReferences()
+            .get(distinct: true)
+            .then((value) => value.length),
+        completion(4));
+
+    // Use backreference
+    expect(
+        db.managers.categories
+            .filter((f) => f.todos((f) => f.title.equals("Math Homework")))
+            .withReferences()
+            .getSingle()
+            .then((value) => value.$1.description),
+        completion("School"));
+
+    // Nested backreference
+    expect(
+        db.managers.categories
+            .filter((f) => f.todos((f) {
+                  final q =
+                      f.category.todos((f) => f.title.equals("Math Homework"));
+                  return q;
+                }))
+            .withReferences()
+            .getSingle()
+            .then((value) => value.$1.description),
+        completion("School"));
+  });
+
+  test('manager - with references tests', () async {
+    // Get department for the 1st product
+    expect(
+        db.managers.product
+            .withReferences()
+            .get(distinct: true)
+            .then((value) =>
+                value.first.$2.department?.getSingle() ?? Future.value(null))
+            .then(
+              (value) => value?.id,
+            ),
+        completion(departments[0].id));
```

Comment:
I know we're also doing this for the other tests, but if we don't need to test concurrent invocations explicitly, `expect(await future, assertion)` is generally more readable than `expect(future, completion(assertion))`.